src/form.ts

```diff
@@ -0,0 +1,173 @@
+const EXCLUDED_TAGS = /\b(?:file|submit|image|reset|button)\b/;
+
+type FormValue = { [ key: string ]: any };
+
+/**
+ * Fills in a DOM form using values from a JavaScript object. Note that fields not specified in the value object will be
+ * cleared.
+ *
+ * @param form the DOM form to set values in
+ * @param object an object containing values for each field in the form
+ *
+ * @example
+ * let formNode = document.getElementById('aForm');
+ * form.fromObject(formNode, { firstName: 'foo', lastName: 'bar' });
+ */
+export function fromObject(form: HTMLFormElement, object: FormValue): void {
+	const elements = form.elements;
+	for (let i = 0; i < elements.length; i++) {
+		const element = <HTMLInputElement> elements[i];
+		const type = element.type;
+		const name = element.name;
+
+		if (!name || EXCLUDED_TAGS.test(type) || element.disabled) {
+			continue;
+		}
+
+		switch (type) {
+			case 'checkbox':
+				element.checked = false;
+				if (name in object) {
+					const checkValue = Array.isArray(object[name]) ? object[name] : [ object[name] ];
+					element.checked = checkValue.indexOf(element.value) !== -1;
+				}
+				break;
+
+			case 'radio':
+				element.checked = (name in object) && object[name] === element.value;
+				break;
+
+			case 'select-multiple':
+				const multiSelectElement: HTMLSelectElement = <any> element;
```

Comment:
The wonkiness necessary here is unfortunate, and I imagine it's because we're casting to `HTMLInputElement` above for the majority of cases which ends up being wrong for this one and the next.  But it's a toss-up between this and having to cast in every single one of them if we started with `HTMLElement`...